src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1114,6 +1117,77 @@ int esbmc_parseoptionst::doit_incremental()
   return 0;
 }
 
+int esbmc_parseoptionst::doit_incremental_cb()
+{
+  optionst opts;
+  get_command_line_options(opts);
+
+  if(get_goto_program(opts, goto_functions))
+    return 6;
+
+  if(cmdline.isset("show-claims"))
+  {
+    const namespacet ns(context);
+    show_claims(ns, get_ui(), goto_functions);
+    return 0;
+  }
+
+  if(set_claims(goto_functions))
+    return 7;
+
+  // Get the initial context bound
+  unsigned int initial_context_bound =
+    strtoul(cmdline.getval("initial-context-bound"), nullptr, 10);
+
+  // Get max number of context bounds
+  unsigned int max_context_bound =
+    cmdline.isset("unlimited-context-bound")
+      ? UINT_MAX
+      : strtoul(cmdline.getval("max-context-bound"), nullptr, 10);
+
+  // Get the context bound increment
+  unsigned int context_bound_inc =
+    strtoul(cmdline.getval("context-bound-step"), nullptr, 10);
+
+  // Get the unwind bound
+  unsigned unwind = atoi(opts.get_option("unwind").c_str());
+
+  bool unknown = true;
+
+  for(unsigned int context_bound = initial_context_bound;
+      context_bound <= max_context_bound;
+      context_bound += context_bound_inc)
+  {
+    opts.set_option("context-bound", integer2string(context_bound));
+    std::cout << "\n*** Context bound number ";
+    std::cout << context_bound;
+    std::cout << " ***\n";
+
+    // If we have found a property violation, then
+    // we stop the verification and report that violation.
+    if(do_base_case(opts, goto_functions, unwind))
+      return true;
+
+    // If we report verification successfully,
+    // we still need to check the other context bounds
+    // since the bug can be exposed to higher context bounds.
+    if(!do_forward_condition(opts, goto_functions, unwind))
+      unknown = false;
+  }
+
+  // If we have explored all context bounds
+  // and did not find a property violation or
+  // the unwinding assertion does not hold,
+  // then we report verification unknown.
+  if(unknown)
+  {
+    status("Unable to prove or falsify the program, giving up.");
+    status("VERIFICATION UNKNOWN");
+  }
+
```

Comment:
The use of the `unknown` variable still looks confusing. 

We are repeatedly calling the forward condition step, without changing the number of loop unwinding. Could you add an example where changing the number of context switches makes the forward condition report the wrong result?

Furthermore, we only set `unknown` to false if the forward condition is false, if it is true, shouldn't we set `unknown` to true? I mean, if for context bound = 2 the forward condition set `unknown` to false, but in context bound = 3 the forward condition doesn't hold, we are still reporting that the verification is not unknown.

Either unknown should be part of the branch condition to call the forward condition or the forward condition should be performed after we finished incrementing the context bound.